storage/innobase/include/buf0dblwr.h

```diff
@@ -195,6 +195,13 @@ struct parallel_dblwr_shard_t {
   buf_page_t **buf_block_arr;
   /** I/O for a doublewrite batch completion event */
   os_event_t batch_completed;
+
+  void reset() {
+    first_free = batch_size = 0;
+    write_buf_unaligned = write_buf = nullptr;
+    buf_block_arr = nullptr;
+    os_event_reset(batch_completed);
```

Comment:
`os_event_reset` acquires a mutex inside `os_event_t`, so we need to double check if `batch_completed`'s constructor was properly called (and therefore that mutex was properly created).
There is a chance (like in a number of places in InnoDB) that we never call `parallel_dblwr_shard_t`'s constructor - just operating on a row block of memory. That is why it used to be OK to do `memset()` on it.